pkg/services/object/server.go

```diff
@@ -182,13 +214,27 @@ func (s *server) Delete(ctx context.Context, req *protoobject.DeleteRequest) (*p
 	}
 
 	t := time.Now()
+	if err := signature.VerifyServiceMessage(delReq); err != nil {
+		return s.makeStatusDeleteResponse(t, err)
+	}
 	resp, err := s.srv.Delete(ctx, delReq)
-	s.pushOpExecResult(stat.MethodObjectDelete, err, t)
 	if err != nil {
-		return nil, err
+		return s.makeStatusDeleteResponse(t, err)
 	}
 
-	return resp.ToGRPCMessage().(*protoobject.DeleteResponse), nil
+	return s.signDeleteResponse(t, util.StatusOKErr, resp.ToGRPCMessage().(*protoobject.DeleteResponse))
+}
+
+func (s *server) signHeadResponse(err error, startTime time.Time, resp *protoobject.HeadResponse) (*protoobject.HeadResponse, error) {
+	resp = util.SignResponse(&s.signer, resp, v2object.HeadResponse{})
+	s.pushOpExecResult(stat.MethodObjectHead, err, startTime)
```

Comment:
Not something I'd expect from `sign...` method.

Reply:
yeah, this is some kind of `finExec`. Ok will refactor